### ISSUE TITLE
Loops in Keras backend.

### ISSUE THREAD

**Issue by dbonadiman** [fchollet/keras#1017]
I'm searching for the discussion somewhere but if i remember correctly somewhere in this video https://www.youtube.com/watch?v=90-S1M7Ny_o Jeff Dean talked about the fact that TensorFlow do not support loops, they basically unfold the scan each time. I think we must do the same for our backend if we don't want fragmentation.

In theano should be possible to unfold a scan. and so in tensor flow. In my opinion we should do the same.

**Comment by EderSantana** [comment]
btw, I updated my TF LSTM example to be independent of batch size. The price was an initial hidden state needing to be defined as input, which I believe is not too bad. https://github.com/EderSantana/TwistedFate/blob/master/mnist_lstm.py

**Comment by yourtheron** [comment]
I am going to write a loss function in which a piece-wise differentiable function will have to be implemented, that is, converting different element in y_true into another value depending on its own value. Is therefore it possible to implement a for-loop in tf backend?
